VRDR/DeathRecord.cs

```diff
@@ -7388,6 +7758,1877 @@ public string EmergingIssue20
+                {
+                    foreach(Observation ob in EntityAxisCauseOfDeathObs)
+                    {
+                        CodeableConcept valueCC = (CodeableConcept)ob.Value;
+                        string value = valueCC.Coding[0].Code;
+                        Observation.ComponentComponent positionComp = ob.Component.Where(c => c.Code.Coding[0].Code=="position").FirstOrDefault();
+                        string position = positionComp.Value.ToString();
+                        Observation.ComponentComponent lineNumComp = ob.Component.Where(c => c.Code.Coding[0].Code=="lineNumber").FirstOrDefault();
+                        string lineNumber = lineNumComp.Value.ToString();
+                        Observation.ComponentComponent codeComp = ob.Component.Where(c => c.Code.Coding[0].Code=="eCodeIndicator").FirstOrDefault();
+                        string code = codeComp.Value.ToString();
+                        Tuple<string, string, string, string> eacod = Tuple.Create(lineNumber, position, value, code);
+                        eac.Add(eacod);
+                    }
+                }
+                return eac.ToArray();
+            }
+            set
+            {
+                // clear all existing eac
+                Bundle.Entry.RemoveAll( entry => entry.Resource.ResourceType == ResourceType.Observation && (((Observation)entry.Resource).Code.Coding.First().Code == "80356-9"));
+                if (EntityAxisCauseOfDeathObs != null)
+                {
+                    EntityAxisCauseOfDeathObs.Clear();
+                }
+                else
+                {
+                    EntityAxisCauseOfDeathObs = new List<Observation>();
+                }
+                // Rebuild the list of observations
+                foreach(Tuple<string, string, string, string> eac in value)
+                {
+                    string lineNumber = eac.Item1;
+                    string position = eac.Item2;
+                    string val = eac.Item3;
+                    string ecode = eac.Item4;
+
+                    Observation ob = new Observation();
+                    ob.Id = Guid.NewGuid().ToString();
+                    ob.Meta = new Meta();
+                    string[] entityAxis_profile = { ProfileURL.EntityAxisCauseOfDeath };
+                    ob.Meta.Profile = entityAxis_profile;
+                    ob.Code = new CodeableConcept(CodeSystems.LOINC, "80356-9", "Cause of death entity axis code [Automated]", null);
+                    ob.Subject = new ResourceReference("urn:uuid:" + Decedent.Id);
+                    AddReferenceToComposition(ob.Id, "CodedContent");
+
+                    ob.Effective = new FhirDateTime();
+                    ob.Value = new CodeableConcept(CodeSystems.ICD10, val, null, null);
+
+                    Observation.ComponentComponent lineNumComp = new Observation.ComponentComponent();
+                    lineNumComp.Value = new FhirString(lineNumber);
```

Comment:
lineNumber and position are not FhirStrings, they're of type Integer

Reply:
updated